src/ts/props/button.ts

```diff
@@ -0,0 +1,33 @@
+import { MantineRadius, MantineSize } from "@mantine/core";
+import React from "react";
+
+export interface __CloseButtonProps {
+    /** Controls width and height of the button. Numbers are converted to rem. `'md'` by default. */
+    size?: MantineSize | (string & {}) | number;
+    /** Key of `theme.radius` or any valid CSS value to set border-radius. Numbers are converted to rem. `theme.defaultRadius` by default. */
+    radius?: MantineRadius;
+    /** Sets `disabled` and `data-disabled` attributes on the button element */
+    disabled?: boolean;
+    /** `X` icon `width` and `height`, `80%` by default */
+    iconSize?: number | string;
+    /** Content rendered inside the button, for example `VisuallyHidden` with label for screen readers */
+    children?: React.ReactNode;
+    /** Replaces default close icon. If set, `iconSize` prop is ignored. */
+    icon?: React.ReactNode;
+}
+
+
+export interface __ClearButtonProps {
+    /** Size of the button, by default value is based on input context */
+    size?: MantineSize | (string & {});
+    /** Key of `theme.radius` or any valid CSS value to set border-radius. Numbers are converted to rem. `theme.defaultRadius` by default. */
+    radius?: MantineRadius;
+    /** Sets `disabled` and `data-disabled` attributes on the button element */
+    disabled?: boolean;
+    /** `X` icon `width` and `height`, `80%` by default */
+    iconSize?: number | string;
+    /** Content rendered inside the button, for example `VisuallyHidden` with label for screen readers */
+    children?: React.ReactNode;
+    /** Replaces default close icon. If set, `iconSize` prop is ignored. */
```

Comment:
Clear buttons still have a default "close" icon?

Reply:
I'm kinda guessing on the clear button props because they aren't listed in their docs

https://mantine.dev/core/input/#inputclearbutton-component
https://mantine.dev/core/input/?t=props

Looking at their source code, it looks like they are taking the CloseButton and just updating it for the calculated size:
https://github.com/mantinedev/mantine/blob/master/packages/%40mantine/core/src/components/Input/InputClearButton/InputClearButton.tsx